pkg/streamingpromql/operators/aggregation.go

```diff
@@ -152,61 +156,100 @@ func (a *Aggregation) NextSeries(ctx context.Context) (types.InstantVectorSeries
 		thisSeriesGroup := a.remainingInnerSeriesToGroup[0]
 		a.remainingInnerSeriesToGroup = a.remainingInnerSeriesToGroup[1:]
 
-		if thisSeriesGroup.sums == nil {
+		if len(s.Floats) > 0 && thisSeriesGroup.floatSums == nil {
 			// First series for this group, populate it.
 
-			thisSeriesGroup.sums, err = a.Pool.GetFloatSlice(steps)
+			thisSeriesGroup.floatSums, err = a.Pool.GetFloatSlice(steps)
 			if err != nil {
 				return types.InstantVectorSeriesData{}, err
 			}
 
-			thisSeriesGroup.present, err = a.Pool.GetBoolSlice(steps)
+			thisSeriesGroup.floatPresent, err = a.Pool.GetBoolSlice(steps)
 			if err != nil {
 				return types.InstantVectorSeriesData{}, err
 			}
+			thisSeriesGroup.floatSums = thisSeriesGroup.floatSums[:steps]
+			thisSeriesGroup.floatPresent = thisSeriesGroup.floatPresent[:steps]
+		}
 
-			thisSeriesGroup.sums = thisSeriesGroup.sums[:steps]
-			thisSeriesGroup.present = thisSeriesGroup.present[:steps]
+		if len(s.Histograms) > 0 && thisSeriesGroup.histogramSums == nil {
+			// First series for this group, populate it.
+			thisSeriesGroup.histogramSums, err = a.Pool.GetHistogramPointerSlice(steps)
+			if err != nil {
+				return types.InstantVectorSeriesData{}, err
+			}
+			thisSeriesGroup.histogramSums = thisSeriesGroup.histogramSums[:steps]
 		}
 
 		for _, p := range s.Floats {
 			idx := (p.T - start) / interval
-			thisSeriesGroup.sums[idx] += p.F
-			thisSeriesGroup.present[idx] = true
+			thisSeriesGroup.floatSums[idx] += p.F
+			if !thisSeriesGroup.floatPresent[idx] {
+				thisSeriesGroup.floatPointCount++
+			}
+			thisSeriesGroup.floatPresent[idx] = true
+		}
+
+		for _, p := range s.Histograms {
+			idx := (p.T - start) / interval
+			if thisSeriesGroup.histogramSums[idx] == nil {
+				// We copy here because we modify the histogram through Add later on.
+				// It is necessary to preserve the original Histogram in case of any range-queries using lookback.
```

Comment:
This is a really good catch. Could you please add a note to `InstantVectorSeriesData.Histograms` that points may share the same `FloatHistogram` pointer and so may not be safe to modify directly?